bug(ec2): `security_group_rule` bindings are incorrectly generated

When creating a `SecurityGroup` with `CDK` and adding a rule using the higher level methods from `Connections`, `cdktf` fails to `synth` because of missing required arguments. The issue is that those arguments are not missing, they are just incorrectly bound.

## Expected Behavior

`SecurityGroup` rules created with `cdk` to be mapped and synthesised correctly, so the generated template can contain the correct configuration.

## Current Behavior

`SecurityGroup` rules are added but are not mapped correctly causing `synth` to fail.

## Steps to Reproduce

I created a sample `cdktf` project to test the functionality of the `adapter`, with one simple `Stack`. You can find the sample code below:

```ts
import { Construct } from "constructs";
import { TerraformStack } from "cdktf";
import { AwsProvider } from "@cdktf/provider-aws/lib/provider";
import * as ec2 from "aws-cdk-lib/aws-ec2";
import { AwsTerraformAdapter } from "@cdktf/aws-cdk";

export class AppsStack extends TerraformStack {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    new AwsProvider(this, "AWS");
    const awsAdapter = new AwsTerraformAdapter(this, "adapter");

    const testVpc = new ec2.Vpc(awsAdapter, "TestVpc");

    const testLoadBalancerSg = new ec2.SecurityGroup(
      awsAdapter,
      "TestAlbSecurityGroup",
      {
        vpc: testVpc,
      },
    );

    const testAppSg = new ec2.SecurityGroup(
      awsAdapter,
      "TestAppSecurityGroup",
      {
        vpc: testVpc,
      },
    );

    testAppSg.connections.allowFrom(testLoadBalancerSg, ec2.Port.tcp(8080));
  }
}

```

Running `cdktf synth` on the above code, will fail due to some missing required arguments, for ex:

```sh
[2023-04-10T14:53:16.692] [ERROR] default - Error: Missing required argument on cdk.tf.json line 240, in resource.aws_security_group_rule.adapter_TestAppSecurityGroupfromadapterTestAlbSecurityGroup14AE107780801BF78829_41D4A789:240: } The argument "security_group_id" is required, but no definition was found.
The argument "protocol" is required, but no definition was found.
The argument "type" is required, but no definition was found.
```

## Detailed Description

The issue seems to be connected to the difference on how `CDK` / `CloudFormation` and `Terraform` security group rules are configured. Some of the incorrect configuration properties that were blocking the `synth`:
1. https://github.com/hashicorp/cdktf-aws-cdk/blob/main/src/aws/security-group-rule/index.ts#L113 - In the `CloudFormation` template this is `ipProtocol`.
2. https://github.com/hashicorp/cdktf-aws-cdk/blob/main/src/aws/security-group-rule/index.ts#L114 - In the `CloudFormation` template this is just `groupId`.
3. https://github.com/hashicorp/cdktf-aws-cdk/blob/main/src/aws/security-group-rule/index.ts#L118 - This one is missing in `CloudFormation`. This can be discovered by the rule configuration, if `sourceSecurityGroupId` is defined, we can treat it as `ingress`, else if `destinationSecurityGroupId` is defined, then we can assume it is `egress`

I did manage to implement the above fixes and the `synth` passed, but since that files is autogenerated by the `cdktf get`, the changes will be deleted. How can this be fixed persistently?

--- a/src/mapping/aws/ec2-vpc.ts
+++ b/src/mapping/aws/ec2-vpc.ts
@@ -116,7 +116,13 @@ registerMapping("AWS::EC2::SecurityGroup", {
 });
 
 registerMapping("AWS::EC2::SecurityGroupEgress", {
-  resource: createGuessingResourceMapper(SecurityGroupRule), // FIXME: create egress rule
+  resource: (scope, id, props) => {
+    props.Type = "egress";
+    return createGuessingResourceMapper(SecurityGroupRule, {
+      GroupId: "securityGroupId",
+      IpProtocol: "protocol",
+    })(scope, id, props);
+  },
   attributes: {
     Arn: () => {
       throw new Error("SecurityGroupRule has no arn");
@@ -126,7 +132,13 @@ registerMapping("AWS::EC2::SecurityGroupEgress", {
 });
 
 registerMapping("AWS::EC2::SecurityGroupIngress", {
-  resource: createGuessingResourceMapper(SecurityGroupRule), // FIXME: create ingress rule
+  resource: (scope, id, props) => {
+    props.Type = "ingress";
+    return createGuessingResourceMapper(SecurityGroupRule, {
+      GroupId: "securityGroupId",
+      IpProtocol: "protocol",
+    })(scope, id, props);
+  },
   attributes: {
     Arn: () => {
       throw new Error("SecurityGroupRule has no arn");

